From afd0be7299533bb2e2b09104399d8a467ecbd2c5 Mon Sep 17 00:00:00 2001
Message-Id: <afd0be7299533bb2e2b09104399d8a467ecbd2c5.1649662041.git.geliang.tang@suse.com>
From: Ciara Loftus <ciara.loftus@intel.com>
Date: Thu, 8 Apr 2021 05:20:09 +0000
Subject: [PATCH] libbpf: Fix potential NULL pointer dereference
Git-commit: afd0be7299533bb2e2b09104399d8a467ecbd2c5
Patch-mainline: v5.12-rc8
References: bsn#127
Modified-by-SEL: No


Wait until after the UMEM is checked for null to dereference it.

Fixes: 43f1bc1efff1 ("libbpf: Restore umem state after socket create failure")
Signed-off-by: Ciara Loftus <ciara.loftus@intel.com>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Link: https://lore.kernel.org/bpf/20210408052009.7844-1-ciara.loftus@intel.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/lib/bpf/xsk.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/tools/lib/bpf/xsk.c
+++ b/tools/lib/bpf/xsk.c
@@ -865,6 +865,8 @@ int xsk_socket__create_shared(struct xsk
 
 	unmap = umem->fill_save != fill;
 
+	unmap = umem->fill_save != fill;
+
 	xsk = calloc(1, sizeof(*xsk));
 	if (!xsk)
 		return -ENOMEM;
